internal/install/recipes/embedded_recipe_fetcher.go

```diff
@@ -0,0 +1,83 @@
+package recipes
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/newrelic/newrelic-cli/internal/install/types"
+)
+
+const (
+	embeddedRecipesPath = "files"
+)
+
+var (
+	//go:embed files/*
+	EmbeddedFS embed.FS
+)
+
+type EmbeddedRecipeFetcher struct{}
+
+func NewEmbeddedRecipeFetcher() *EmbeddedRecipeFetcher {
+	return &EmbeddedRecipeFetcher{}
+}
+
+func (f *EmbeddedRecipeFetcher) FetchRecipes(context.Context) (out []types.OpenInstallationRecipe, err error) {
+	files, err := f.getYAMLFiles(embeddedRecipesPath)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, f := range files {
+		b, err := EmbeddedFS.ReadFile(f)
+		if err != nil {
+			return nil, fmt.Errorf("could not read embedded file %s: %w", f, err)
+		}
+
+		var r types.OpenInstallationRecipe
+		if err := yaml.Unmarshal(b, &r); err != nil {
+			return nil, fmt.Errorf("could not unmarshal embedded file %s: %w", f, err)
+		}
+
+		out = append(out, r)
+	}
+
+	return out, nil
+}
+
+func (f *EmbeddedRecipeFetcher) getYAMLFiles(path string) (out []string, err error) {
+	return f.getFiles(path, isYAMLFile)
+}
+
+func isYAMLFile(path string) bool {
+	e := filepath.Ext(path)
+	return e == ".yml" || e == ".yaml"
```

Comment:
@ctrombley maybe a case insensitive compare, just in case for windows?

Reply:
👍